### PR TITLE
core(main-thread-tasks): handle last task not finishing

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Treats `'B'` events without a corresponding `'E'` as if they ended at the time of trace end.

**Related Issues/PRs**
#7691

### CODE REVIEW

**Comment by Hoten** [comment]
Looks good. I'll verify in Canary now.

File: lighthouse-core/computed/main-thread-tasks.js [hunk 1]
```diff
@@ -81,8 +81,13 @@ class MainThreadTasks {
     const tasks = [];
     /** @type {TaskNode|undefined} */
     let currentTask;
+    let lastTraceTs = 0;
 
     for (const event of mainThreadEvents) {
+      // Keep track of the last trace event timestamp
+      if (event.ts > lastTraceTs) lastTraceTs = event.ts;
+      if (currentTask && currentTask.endTime > lastTraceTs) lastTraceTs = currentTask.endTime;
```
Comment (brendankenny) [inline lighthouse-core/computed/main-thread-tasks.js]: rather than doing this (which is a hard to follow with the `currentTask` check and the (I assume) `NaN` comparison), could we either pass in `TraceOfTab.timestamps.traceEnd` or (probably less ideal) replicate the logic from there since the events are sorted by ts?

File: lighthouse-core/computed/main-thread-tasks.js [hunk 2]
```diff
@@ -133,6 +138,12 @@ class MainThreadTasks {
       }
     }
 
+    while (currentTask && !Number.isFinite(currentTask.endTime)) {
+      // The last event didn't finish before tracing stopped, just end it when tracing ended.
+      currentTask.endTime = lastTraceTs;
+      currentTask = currentTask.parent;
+    }
+
```
Comment (paulirish) [inline lighthouse-core/computed/main-thread-tasks.js]: Lets add a comment here with our expectation like

    // At this point we expect all tasks to have a finite startTime and endTime.

File: lighthouse-core/computed/main-thread-tasks.js [hunk 3]
```diff
@@ -133,6 +138,12 @@ class MainThreadTasks {
       }
     }
 
+    while (currentTask && !Number.isFinite(currentTask.endTime)) {
```
Comment (paulirish) [inline lighthouse-core/computed/main-thread-tasks.js]: add this sorta comment in here? took me a second to determine what currentTask was at this point.

    // Starting from the last and bottom-most task,
